### PR TITLE
Add unit tests for badge, userbadge models and respective views

### PR DESCRIPTION
**Pull request by mabdullahz**
[`assign_badge`, `get_unearned_badges`] Coverage for openedx/features/badging/models.py : 93%
[`assign_user_badge`] Coverage for lms/djangoapps/philu_api/views.py : 100%

### CODE REVIEW

File: openedx/features/badging/tests/test_model_badge.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        badge_2.save()
+
+        UserBadge.objects.create(
+            user_id=user.id,
+            badge_id=badge_1.id,
+            course_id="",
+            community_id=-1
+        )
+
+        UserBadge.objects.create(
```
Comment (saad-waqar) [inline openedx/features/badging/tests/test_model_badge.py]: Lets create factories

File: lms/djangoapps/philu_api/tests/test_views.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+
+class BadgeAssignViewTest(TestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.assign_path = reverse('assign_user_badge')
+
+    @patch('lms.djangoapps.philu_api.views.UserBadge.assign_badge')
+    def test_assign_user_badge_wrong_token(self, mock_userbadge_assign_badge):
+        dict_data = {
```
[the remark this answers is not in the document]
Comment (mabdullahz) [reply lms/djangoapps/philu_api/tests/test_views.py]: Done.

File: openedx/features/badging/tests/test_model_badge.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+class BadgeModelTestCases(TestCase):
+    def test_name_blank_raises_IntegrityError(self):
+        """
+        Trying to save a Badge object with no name raises
+        an IntegrityError exception.
+        """
+        badge = Badge(name=None,
+                      description="This is a sample badge",
+                      threshold=30,
+                      type="conversationalist",
```
[the remark this answers is not in the document]
Comment (mabdullahz) [reply openedx/features/badging/tests/test_model_badge.py]: Done.

File: openedx/features/badging/tests/test_model_badge.py [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+        """
+        badge = Badge(name="Sample Badge",
+                      description="This is a sample badge",
+                      threshold=30,
+                      type="conversationalist",
+                      image="path/to/image",
+                      date_created=None)
+        self.assertEqual(badge.save(), None)
+
+    def test_remaining_badges_empty(self):
```
[the remark this answers is not in the document]
Comment (mabdullahz) [reply openedx/features/badging/tests/test_model_badge.py]: DOne.

File: openedx/features/badging/tests/test_model_badge.py [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+        actual_result["2"] = actual_result.pop(actual_result.keys()[1])
+
+        self.assertDictEqual(actual_result, expected_result)
+
+    def test_remaining_badges_one_earned(self):
+        """
+        Get remaining badges dictionary after saving badges
+        when user has already earned one of the badges
+        """
+        user = User.objects.create_user(username='testuser', password='12345')
```
[the remark this answers is not in the document]
Comment (mabdullahz) [reply openedx/features/badging/tests/test_model_badge.py]: Done.

File: openedx/features/badging/tests/test_model_badge.py [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+        """
+        badge_1 = Badge(name="badge_1", description="This is a sample badge",
+                        threshold=30, type="team", image="path/to/image")
+        badge_1.save()
+        badge_2 = Badge(name="badge_2", description="This is a sample badge",
+                        threshold=70, type="team", image="path/to/image")
+        badge_2.save()
+
+        expected_result = {}
+        expected_result["1"] = {'name': badge_1.name,
```
[the remark this answers is not in the document]
Comment (mabdullahz) [reply openedx/features/badging/tests/test_model_badge.py]: Done.

File: openedx/features/badging/tests/test_model_badge.py (hunk 1 above)
Comment (mabdullahz) [reply openedx/features/badging/tests/test_model_badge.py]: Done.

**Comment by ahmed-faraz46** [comment]
@mabdullahz do you think if this PR is still valid? Please proceed with it accordingly.

**Comment by mabdullahz** [comment]
> @mabdullahz do you think if this PR is still valid? Please proceed with it accordingly.

@ahmed-faraz46 the unit tests were merged in `philu.ironwood.master`. If we don't plan on moving back to `release-v1` at any point in the future, I think this PR is invalidated and should be closed. What do you think? @saad-waqar input?

**Comment by ahmed-faraz46** [comment]
closing because it was directly merged in philu.ironwood.master and was released under release-v-2.0

**Pull request closed without merging**